manipbolt/compiler.go

```diff
@@ -0,0 +1,101 @@
+package manipbolt
+
+import (
+	"fmt"
+
+	"github.com/asdine/storm/q"
+	"go.aporeto.io/elemental"
+	"go.aporeto.io/manipulate"
+)
+
+func compileFilter(f *elemental.Filter) (q.Matcher, error) {
+
+	if len(f.Operators()) == 0 {
+		return q.And(), nil
+	}
+
+	matchers := []q.Matcher{}
+
+	for i, operator := range f.Operators() {
+
+		switch operator {
+
+		case elemental.AndOperator:
+
+			k := f.Keys()[i]
+			values := f.Values()[i]
+			items := []q.Matcher{}
+
+			switch f.Comparators()[i] {
+
+			case elemental.EqualComparator:
+
+				items = append(items, containsOrEqual(k, values[0]))
```

Comment:
I haven't checked that function yet, but either it is not doing what it says it does (equality check) or should not be used. From what I understand, a == "y" would match x if x == "ay"